### PR TITLE
feature/expiredTask

### PR DESCRIPTION
**Pull request by bsmmoon**
- [x] closes #165 

It will check due dates expiration every time Do Lah is launched.

### CODE REVIEW

File: DoLahUI/UITaskBox.cpp [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace DoLah {
     
     void UITaskBox::dynamicCast(int index, AbstractTask *task) {
         std::string name = task->getName();
+        if (task->isExpired()) {
+            name = "(overdue) " + name;
+        }
```
Comment (bsmmoon) [inline DoLahUI/UITaskBox.cpp]: Can be enhanced by more elegant display.

File: DoLah/Models/Task.cpp [hunk 2]
```diff
@@ -30,6 +30,17 @@ namespace DoLah {
...
+    void AbstractTask::setExpired(bool expired) {
+        this->expired = expired;
+    }
+
+    bool AbstractTask::isExpired() {
+        return this->expired;
+    }
+
+    void AbstractTask::updateExpired() {
+    }
```
Comment (9gix) [inline DoLah/Models/Task.cpp]: @bsmmoon what is this function suppose to do?

File: DoLah/Models/Task.cpp [hunk 3]
```diff
@@ -30,6 +30,17 @@ namespace DoLah {
         return this->tags;
     }
 
+    void AbstractTask::setExpired(bool expired) {
+        this->expired = expired;
+    }
+
+    bool AbstractTask::isExpired() {
+        return this->expired;
+    }
```
Comment (9gix) [inline DoLah/Models/Task.cpp]: This function should be calculate whether it is expiry at the time when it was called, and the value of expiration should be computed directly when calling this function.

File: DoLah/Storage/CalendarStorage.cpp [hunk 4]
```diff
@@ -112,6 +112,7 @@ namespace DoLah {
                 }
             }
         }
+        calendar.updateTaskExpiry();
```
Comment (9gix) [inline DoLah/Storage/CalendarStorage.cpp]: @bsmmoon  Expiry should not be updated upon loaded. but rather should be computed directly using `AbstractTask::isExpire()`

**Comment by 9gix** [comment]
@bsmmoon suggestion for improvement in the comment above. (consider deleting this branch)

**Pull request merged**